.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,54 @@
+name: Benchmarks
+on:
+  schedule:
+    - cron: "0 0 * * 0-6"  # Run daily at 0am
+  pull_request:
+
+concurrency:
+  group: benchmarks-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+    # If needed, this workflow will create a cached benchmark reference.
+    # This reference will be used for comparative analyses with all PRs benchmarks.
+    reference-benchmarks:
+      if: ${{(github.event_name == 'schedule' || (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'ci:run-benchmarks')))}}
+      uses: ./.github/workflows/interface-unit-tests.yml
+      with:
+        branch: add-pytest-benchmarks

```

Comment:
Would this branch stay this value? Or we want to update this to `master` or last stable release (tag) etc?

Reply:
We will be changing that in the future. For now, we are using this branch as a reference.